### PR TITLE
GUVNOR-2290 - Allow custom widgets on primary navbar and custom posit…

### PR DESCRIPTION
**Pull request by cristianonicolai**
…ion for perspective context menu

 Custom widget on primary navbar, required for allowing the creating of custom menu items like search input
 Using MenuPosition for perspective context menu so that items can be aligned to right side.

### CODE REVIEW

File: uberfire-showcase/uberfire-webapp/src/main/java/org/uberfire/client/navbar/SearchMenuBuilder.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+    private AnchorListItem link = GWT.create( AnchorListItem.class );
+
+    public SearchMenuBuilder() {
+        link.setIcon( IconType.SEARCH );
+        link.setPull( Pull.RIGHT );
+        link.addClickHandler( new ClickHandler() {
+            @Override
+            public void onClick( ClickEvent event ) {
+                Window.alert( "Search!" );
```
Comment (manstis) [inline uberfire-showcase/uberfire-webapp/src/main/java/org/uberfire/client/navbar/SearchMenuBuilder.java]: Are you sure this is correct?

File: uberfire-workbench/uberfire-workbench-client-views-patternfly/src/main/java/org/uberfire/client/views/pfly/menu/WorkbenchMenuStandardNavBarView.java [hunk 2]
```diff
@@ -98,6 +111,10 @@ public void onClick( ClickEvent event ) {
             } );
         }
 
+        if( position == MenuPosition.RIGHT ){
```
Comment (manstis) [inline uberfire-workbench/uberfire-workbench-client-views-patternfly/src/main/java/org/uberfire/client/views/pfly/menu/WorkbenchMenuStandardNavBarView.java]: What about ```LEFT``` and ```CENTER```?

**Comment by manstis** [comment]
Other than a couple of minor (and probably explainable) comments; +1 to merge.

File: uberfire-showcase/uberfire-webapp/src/main/java/org/uberfire/client/navbar/SearchMenuBuilder.java (hunk 1 above)
Comment (cristianonicolai) [inline uberfire-showcase/uberfire-webapp/src/main/java/org/uberfire/client/navbar/SearchMenuBuilder.java]: yeah, just my lack of creativity for the showcase :)

File: uberfire-workbench/uberfire-workbench-client-views-patternfly/src/main/java/org/uberfire/client/views/pfly/menu/WorkbenchMenuStandardNavBarView.java (hunk 2 above)
Comment (cristianonicolai) [inline uberfire-workbench/uberfire-workbench-client-views-patternfly/src/main/java/org/uberfire/client/views/pfly/menu/WorkbenchMenuStandardNavBarView.java]: only RIGHT needs to be pushed with a float, LEFT and CENTER just follow the other menus are added.

**Comment by manstis** [comment]
+1 to merge.

**Comment by csadilek** [comment]
master: 6138d9de9d506103ccd2c65e4ea9e7efc0322cdf

**Pull request closed without merging**